EducationalAssignments/DefaultPartOne.md

```diff
@@ -0,0 +1,302 @@
+This defense monitor will ensure that software follows certain rules and guidelines, preventing potential unauthorized actions.
+
+You should write test applications to ensure your reference monitor behaves
+properly in different cases and to test attacks against your monitor.    
+
+
+### Specifications:
+
+1. Your defense monitor should incorporate all the standard file operation methods, from opening a file, reading and writing to it, to closing and deleting it. 
+
+2. In addition, a specially named file - `default` - shall be used as a template when opening files without creating them first.
+Eg: Calling `openfile('foo', True)` should create and open a new empty file called `foo` (assuming it's not present already). However, calling `openfile('foo', False)` should create a new file using `default` as the template. If `default` doesn't exist, throw the relevant error (`FileNotFoundError`).
+
+3. If default is created, written to or deleted, then all closed files that were previously created gets deleted. Any files that are already open must be left unchanged.
+
+Three design paradigms are at work in this assignment: accuracy, efficiency, and
+security.
+
+* Accuracy: The defense monitor should precisely and consistently manage file
+operations. Only specific operations, such as `openfile()`, should have their
+behavior modified. All situations that are not described above *must* match that
+of the underlying API.
+
+* Efficiency: The security layer should use a minimum number of resources, so
+performance is not compromised.  For example, you may not call `readat()`
+everytime `writeat()` is called.  It is permissable to call `readat()` upon
+`fileopen()`, however.
+
+ * Security: The defense layer should be robust against tampering and
+circumvention. Attackers must not be able to bypass, disable, or exploit the
+defense monitor's enhanced behaviors, ensuring the integrity and intended
+functionality of file operations.
+
+
+### Getting Python and RepyV2
+
+Please refer to the [SeattleTestbed Build
+Instructions](../Contributing/BuildInstructions.md#prerequisites) for details.
+
+Once you have built RepyV2 into a directory of your choice, change into that
+directory. Use the command below in order to run your RepyV2 applications:
+
+```bash
+python2 repy.py restrictions.default encasementlib.r2py [security_layer].r2py [application].r2py
+```
+
+(Replace `[security_layer].r2py` and `[application].r2py` by the names of the
+security layers and application that you want to run.) 
+
+In order to test whether or not these steps worked, please copy and paste the
+code found below for the sample security layer and sample attack.
+
+You should get a `FileNotFoundError` when running the test.  If not, please go
+through the troubleshooting section found below.
+
+
+### Troubleshooting Repy code
+
+If you can't get Repy files to run, some of the following common errors may have
+occurred:
+
+ * using `print` instead of `log`:
```

Comment:
```suggestion
 * using `log` instead of `print`:
```

Reply:
The statement before this says, "some of the following common errors may have occurred:". So using 'print' instead of 'log' is the error right ?